sillsdev/LfMerge
# Don't serialize comments to JSON when calling LfMergeBridge

Fixes #329.

Requires https://github.com/sillsdev/flexbridge/pull/401 and https://github.com/sillsdev/flexbridge/pull/402 to be built into a NuGet package before this can fully work. At the moment, those PRs have been merged into the flexbridge `develop` branch but the NuGet package build for FLExBridge is not yet working. So just for now, this only works if you check out the https://github.com/sillsdev/flexbridge repo to a sibling folder of LfMerge. (LfMerge currently uses a ProjectReference for flexbridge in order to be able to run the newly-merged code). Once a new NuGet package is built from flexbridge's `develop` branch, this PR can switch from using a ProjectReference for flexbridge to using a proper PackageReference, and then there won't be any need to check out the flexbridge repo alongside the lfmerge one.

Currently, PR checks are going to fail because Ci isn't set up to clone flexbridge alongside lfmerge, so the ProjectReference will fail. That's expected, and is one reason why https://github.com/sillsdev/flexbridge/pull/402 has to be merged first before this PR is merged.

## To test:

1. Make sure you have Language Forge and LexBox running side by side.
2. Check out `flexbridge` repo in a sibling folder to `LfMerge`.
3. In LfMerge repo, run `./pbuild.sh`. (Not yet renamed to `./build.sh` as of when this PR was branched off from `develop`).
4. Note the Docker image tag produced (probably something like `2.0.140-feat-less-serialization-of-comments.10`) and copy it to the clipboard.
5. In the web-languageforge repo, edit `docker/lfmerge/Dockerfile`
6. Change the lfmerge tag to the one from step 4.
7. In the web-languageforge repo, run `make`.
8. Run `docker logs -f lfmerge` so you'll see if there are any Send/Receive errors logged in the next steps.
9. Go to http://localhost/project/sena-3 (LexBox) and look at the project history of sena-3
10. Go to http://localhost:8080/ (Language Forge) in another tab and clone the sena-3 project.
11. Edit the sena-3 project in Language Forge and add a comment to an entry. Do a Send/Receive.
12. After the Send/Receive completes, go to http://localhost/project/sena-3 and verify that there's a new commit in sena-3.
13. Click on the new commit to be taken to the Mercurial repo for sena-3.
14. Click on `log` in the Mercurial repo for sena-3.
15. Click on the latest commit, which should be the one created by the Language Forge S/R.
16. Verify that it adds a message to the `.ChorusNotes` file, creating it if it didn't already exist.

That's a lot of steps, sorry. But this is just inherently complex; once a FlexBridge NuGet package has been built, you'll be able to skip steps 1 and 2, but the rest of the steps are necessary to test this feature.
